File: Abstraction/Parametric/abs.java
class Main {
    public static <T extends Number & Comparable<T>> T abs(T arg) {
        return (arg.doubleValue() > 0) ? arg : (T) Double.valueOf(-arg.doubleValue());
    }

    public static void main(String[] args) {
        System.out.println(abs(4));
        System.out.println(abs(-3.6));
    }
}

File: Abstraction/Parametric/abs.kt
@Suppress("UNCHECKED_CAST")
fun <T> abs(arg: T): T where T: Number, T: Comparable<T> {
    return if (arg.toDouble() > 0) arg else -arg.toDouble() as T
}

fun main() {
    println(abs(4))
    println(abs(-3.6))
}
